cuda_core/tests/test_memory.py

```diff
@@ -288,13 +279,18 @@ def test_buffer_dunder_dlpack_device_failure():
 
 
 @pytest.mark.parametrize("use_device_object", [True, False])
-def test_device_memory_resource_initialization(mempool_device, use_device_object):
+def test_device_memory_resource_initialization(use_device_object):
     """Test that DeviceMemoryResource can be initialized successfully.
 
     This test verifies that the DeviceMemoryResource initializes properly,
     including the release threshold configuration for performance optimization.
     """
-    device = mempool_device
+    device = Device()
+
+    if not device.properties.memory_pools_supported:
+        pytest.skip("Device does not support mempool operations")
+
+    device.set_current()
```

Comment:
Q: Any reason we don't wanna move this logic (check attribute first, and then either skip or set current) in the `mempool_device` fixture, and then use the fixture here?